test/lib/clients/repositories_test.rb

```diff
@@ -46,4 +46,25 @@
       end
     end
   end
+
+  describe "#warn_service_repository" do
+    context "when the repository is managed by a service" do
+      let(:source_state) { { "name" => "repo", "service" => "some-service", "SrcId" => "1" } }
+
+      it "shows a warning message only once" do
+        expect(Yast::Popup).to receive(:Warning).once
```

Comment:
I'd add a parameter, potentially any other warning can be displayed during the test (e.g. a crash popup), just make sure the expected one is displayed: `.with(/manual changes will be reset/)`.